js/glpi_install.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * ---------------------------------------------------------------------
+ *
+ * GLPI - Gestionnaire Libre de Parc Informatique
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2024 Teclib' and contributors.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+let request_running = false;
+
+function startDatabaseInstall(message_element_id, success_html)
+{
+    const message_element = document.getElementById(message_element_id);
+    if (!message_element) {
+        const alert = document.createElement('p');
+        alert.innerText = 'Could not load HtmlElement to append messages to.';
+        document.body.appendChild(alert);
+        throw new Error(alert.innerText);
+    }
+
+    const single_message_element = document.createElement('p');
+    const msg_list_element = document.createElement('div');
+    message_element.appendChild(msg_list_element);
+    msg_list_element.appendChild(single_message_element);
+
+    function message(text) {
+        const alert = document.createElement('p');
+        alert.innerHTML = text;
+        msg_list_element.appendChild(alert);
+    }
+    function queries_message(amount) {
+        single_message_element.innerHTML = `Process: ${amount}`;
+    }
+
+    request_running = true;
+
+    setTimeout(() => {
+        checkProgress(queries_message);
+    }, 1500);
+
+
+    fetch("/install/step7/start_db_inserts", {
+        method: 'POST',
+        headers: {
+            'Content-Type': 'text/plain'
+        },
+    })
+        .then((res) => res.text())
+        .then((text) => {
+            if (text && text.trim().length) {
+                message(`Error:\n${text}`);
+            } else {
+                message(success_html);
+                queries_message('✅');
+            }
+        })
+        .finally(() => request_running = false);
```

Comment:
This request will probably still timeouts if there is a proxy between the client and the PHP server (e.g. nginx + php-fpm), so you cannot rely on its response.
The finished/aborted state should be returned by the `check_progress` endpoint.

Reply:
The `.finally()` method will be always called, and I make sure that the checkProgress function stops when it finds a 404 (meaning progress has ended). This means that it cannot run on an endless loop, it'll always stop at some point.

However, what I can do is add a `.catch()` to make sure the user sees when there's an error.

Reply:
My point is that receiving this request will return an error code if there is a timeout between nginx and php-fpm for instance. It means that we cannot trust the response code for this request, and this is exactly the problem that we are trying to handle with this PR.

The most viable solution is, IMHO, to get the success/failure status from the progress endpoint.